crates/sargon/src/profile/mfa/security_structures/roles/builder/roles_builder.rs

```diff
@@ -629,14 +735,18 @@ impl<const ROLE: u8> RoleBuilder<ROLE> {
                     PrimaryCannotContainTrustedContact,
                 );
             }
-            FactorSourceKind::Device => {
-                if self.contains_factor_source_of_kind(FactorSourceKind::Device)
-                {
-                    return RoleBuilderMutateResult::forever_invalid(
-                        PrimaryCannotHaveMultipleDevices,
-                    );
+            FactorSourceKind::Device => match mode {
+                SecurityShieldBuilderMode::Strict => {
+                    if self.contains_factor_source_of_kind(
+                        FactorSourceKind::Device,
+                    ) {
+                        return RoleBuilderMutateResult::forever_invalid(
+                            PrimaryCannotHaveMultipleDevices,
+                        );
+                    }
                 }
-            }
+                SecurityShieldBuilderMode::Lenient => {}
```

Comment:
for `Lenient` we do not count the number of `Device` FactorSource, we allow many.